doc: add helm supported version note to upgrade doc (#11734)
rook/rook: Documentation/Upgrade/rook-upgrade.md

## Helm

> * Be sure to update to a [supported Helm version](https://helm.sh/docs/topics/version_skew/#supported-version-skew)

Reviewer: How about if we move this to the end of this helm section, and change it to a note? It just doesn't seem critical to list first in the section.
```suggestion
!!! note
 Be sure you are using a [supported Helm version](https://helm.sh/docs/topics/version_skew/#supported-version-skew)
```
